tests/orm/test_fields.py

```diff
@@ -0,0 +1,222 @@
+    data_regression.check(
+        {key: repr(value) for key, value in node_cls.fields._dict.items()},
+        basename=f'fields_{group}.{name}.{node_cls.__name__}',
+    )
+
+
+def test_add_field():
+    """Test the `add_field` API."""
+
+    class NewNode(orm.Data):
+        __qb_fields__ = (
+            add_field(
+                'key1',
+                dtype=str,
+                is_db_field=False,  # `qb_field` will yield "attributes.key1"
+                is_subscriptable=False,
+            ),
+        )
+
+    node = NewNode()
+
+    assert 'key1' in node.fields
+    assert node.fields.key1.dtype is str
+    assert isinstance(node.fields.key1, orm.fields.QbStrField)
+    assert node.fields.key1.qb_field == 'attributes.key1'
+    assert not node.fields.key1.is_subscriptable
+
+
+@pytest.mark.parametrize('key', ('|', 'some.field', '1key'))
+def test_invalid_field_keys(key):
+    """Test for invalid field keys."""
+    with pytest.raises(ValueError):
+        _ = add_field(key)
+
+
+def test_disallowed_alias_for_db_field():
+    """Test for disallowed alias argument for database fields."""
+    with pytest.raises(ValueError):
+        _ = add_field(
+            'some_key',
+            'alias_not_allowed_for_db_fields',
+            is_db_field=True,
+        )
+
+
+def test_query_new_class(clear_database_before_test, monkeypatch):
+    """Test that fields are correctly registered on a new data class,
+    and can be used in a query.
+    """
+    from aiida import plugins
+
+    def _dummy(*args, **kwargs):
+        return True
+
+    monkeypatch.setattr(
+        plugins.entry_point,
+        'is_registered_entry_point',
+        _dummy,
+    )
+
+    class NewNode(orm.Data):
+        __qb_fields__ = [
+            add_field('some_label', dtype=str),
+            add_field('some_value', dtype=int),
+        ]
+
+    node = NewNode()
+    node.base.attributes.set_many({'some_label': 'A', 'some_value': 1})
+    node.store()
+
+    node = NewNode()
+    node.base.attributes.set_many({'some_label': 'B', 'some_value': 2})
+    node.store()
+
+    node = NewNode()
+    node.base.attributes.set_many({'some_label': 'C', 'some_value': 3})
+    node.store()
+
+    result = (
+        orm.QueryBuilder()
+        .append(
+            NewNode,
+            tag='node',
+            project=[
+                NewNode.fields.some_label,
+                NewNode.fields.some_value,
+            ],
+            filters=NewNode.fields.some_value > 1,
+        )
+        .order_by({'node': NewNode.fields.ctime})
+        .all()
+    )
+    assert result == [['B', 2], ['C', 3]]
+
+
+def test_filter_operators():
+    """Test that the operators are correctly registered."""
+    pk = orm.Data.fields.pk
+    filters = (pk == 1) & (pk != 2) & (pk > 3) & (pk >= 4) & (pk < 5) & (pk <= 6) & ~(pk == 7) & ~(pk < 8)
+    # print(filters.as_dict())
+    assert filters.as_dict() == {
+        'and': [
+            {'pk': {'==': 1}},
```

Comment:
@sphuber note that a side effect of the mapping is precisely that we no longer need the backend aliases. Even in string mode, we can use `pk`.